Obsidian/WorldData/World.cs

```diff
@@ -529,9 +529,12 @@ public async Task ManageChunksAsync()
 
         public async Task FlushRegionsAsync()
         {
-            Server.BroadcastMessage("Saving to disk...");
-            Parallel.ForEach(Regions.Values, async r => await r.FlushAsync());
-            Server.BroadcastMessage("Save complete.");
+            await Task.Run(() =>
+            {
+                Server.BroadcastMessage("Saving to disk...");
+                Parallel.ForEach(Regions.Values, async r => await r.FlushAsync());
+                Server.BroadcastMessage("Save complete.");
+            });
```

Comment:
Should possibly use [Task.WhenAll](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.whenall) instead, but I am unsure whether IO blocking could interfere when flushing

Reply:
Implemented the WhenAll